src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
In my opinion, there is too much functionality here. How do you look at first all the children, save them to a variable, and then create a variable in which you already choose whether you need only men among these children using the function of finding people by gender.
And do not abbreviate the names of parameters or variables